src/temp-alarm.ino

```diff
@@ -1,40 +1,45 @@
-#include <Keypad.h>
+// #include <Keypad.h>
 
 #define LM35_LOAD_PIN A7
 #define LM35_SAMPLE_SIZE 128
 #include <LM35.h>
 #include <SSD1306AsciiAvrI2c.h>
 #include <EEPROM.h>
 
-#include <TempAlarmControl.h>
+// #include <TempAlarmControl.h>
 
 #define I2C_ADDRESS 0x3C
 #define RST_PIN -1
 
 #define KEYPAD_ROWS 4
 #define KEYPAD_COLS 4
 
+#define PIN_SCK 13
+#define PIN_MISO 12
+#define PIN_MOSI 11
+#define PIN_SS 10
+#define PIN_SSIN 9
+#define PIN_LOAD 8
+#define PIN_INTERRUPT 3
+
 #define DEBUG
 
-static LM35 temp(1100);
+#include <SPI.h>
+
+LM35 temp(1100);
 
 const unsigned long startTime = millis();
 
-const char hexaKeys[KEYPAD_ROWS][KEYPAD_COLS] = {
-    { '1', '2', '3', 'A' },
-    { '4', '5', '6', 'B' },
-    { '7', '8', '9', 'C' },
-    { '*', '0', '#', 'D' },
+// TODO I could swap row/column wires to make this array "right-side up"
+const char hexaKeys[KEYPAD_ROWS*KEYPAD_COLS] = {

```

Comment:
Is there efficiency to be gained by switching to a one-dimensional array, or is it mostly ａｅｓｔｈｅｔｉｃ?

Reply:
yeah, the key read is the index into this array. I mixed up the wiring which is why I had to transpose it.